docarray/typing/abstract_type.py

```diff
@@ -0,0 +1,30 @@
+from abc import abstractmethod
+from typing import Any, Type, TypeVar, Union
+
+from pydantic import BaseConfig
+from pydantic.fields import ModelField
+
+from docarray.document import BaseNode
+
+T = TypeVar('T')
+
+
+class AbstractType(BaseNode):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    @abstractmethod
+    def validate(
+        cls: Type[T],
+        value: Any,
+        field: ModelField,
+        config: BaseConfig,
```

Comment:
```suggestion
        field: 'ModelField',
        config: 'BaseConfig',
```

these two object are only imported for type testing therefore we need to put them into ' ' otherwise python complain